### PR TITLE
[build] Use --file-alignment=4096 with MinGW

### PR DESCRIPTION
**Pull request by Joshua-Ashton**
Avoids a copy in the Wine loader as well as enables debug symbols to work in perf.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -5,15 +5,16 @@ cpu_family = target_machine.cpu_family()
 add_project_arguments('-DNOMINMAX', language : 'cpp')
 
 dxvk_compiler = meson.get_compiler('cpp')
-dxvk_msvc=dxvk_compiler.get_id() == 'msvc'
+dxvk_msvc = dxvk_compiler.get_id() == 'msvc'
```
Comment (doitsujin) [inline meson.build:8]: can we name this `is_msvc` or something?

Comment (Joshua-Ashton) [reply meson.build]: Done.

Comment (Joshua-Ashton) [reply meson.build]: wait

Comment (Joshua-Ashton) [reply meson.build]: Forgot to add a few files to the commit, fixed.

**Comment by pchome** [comment]
Is `objects             : not dxvk_is_msvc ? 'd3d10'+def_spec_ext : [],` still required by anything? It was a hack for winelib to add *.spec file name to link process, IIRC. I guess I about to add additional headache for me and other winelib users, but better remove this if unused.

**Comment by Joshua-Ashton** [comment]
Its needed for MSVC to compile

**Comment by pchome** [comment]
Likely `vs_module_defs      : 'd3d10'+def_spec_ext,`  needed for MSVC to compile. The question is this also used for MinGW, or mingw use `objects`.

**Comment by Joshua-Ashton** [comment]
MinGW takes in it's def as an object, MSVC takes it in another way, hence why this is needed.

**Comment by pchome** [comment]
Still doubt, but okay I won't insist.

DXVK was fine w/o this.
- then they added msvc support: https://github.com/doitsujin/dxvk/commit/9deb73a2a7007c20d306f2e88d783458b4935825#diff-02d517bb3be49b91036dde4a613d0b4b5793091470cb8e3f51f01ee1ce7f0218
- then winelib support was added and broke msvc: https://github.com/doitsujin/dxvk/commit/06511aa72c05538884b937227ecfa0ab041fe854#diff-02d517bb3be49b91036dde4a613d0b4b5793091470cb8e3f51f01ee1ce7f0218
(based on my PR https://github.com/doitsujin/dxvk/pull/268/files#diff-02d517bb3be49b91036dde4a613d0b4b5793091470cb8e3f51f01ee1ce7f0218, so it was literally me who introduced this shit (`object:`))
- then msvc was fixed: https://github.com/doitsujin/dxvk/commit/f38ee85a396bbc2a0ad2dba8f14f5a34d61457ee#diff-02d517bb3be49b91036dde4a613d0b4b5793091470cb8e3f51f01ee1ce7f0218

That's why I doubt. I don't have all compilers to check, but I bet you can remove this (`object:`) to clean up a bit.

**Comment by Joshua-Ashton** [comment]
MinGW does not respect vs_module_defs, MSVC fails if you have the .def in the object array.

I don't get what you don't get.

**Comment by pchome** [comment]
I don't get why you need `objects` for MSVC if it is empty `[]` for MSVC. And if you do not need `objects` not for MSVC nor for MinGW then remove it, because it is not intended to be used for def/spec anyway.

**Comment by Joshua-Ashton** [comment]
This is where you put .def on MinGW. Unless something has changed, it does not respect vs_module_defs as I mentioned above.

**Pull request merged**